experimental/Schemes/CoveredScheme.jl

```diff
@@ -146,22 +146,22 @@ function intersect_in_covering(U::AbsSpec, V::AbsSpec, C::Covering)
   (l, m, n) = indexin(V, C)
   if i == l # U and V are affine opens of the same patch X in C
     X = C[i]
-    if X == U == V 
-      iso = identity_map(SpecOpen(X))
+    if X == U == V
+      iso = identity_map(X)
       return iso, iso, iso, iso
     end
-    f = one(base_ring(OO(X))) # For the case where U is already a basic patch
+    f = one(ambient_ring(X)) # For the case where U is already a basic patch
     if j != 0 # In this case, U appears in some refinement
-      f = gens(affine_refinements(C)[C[i]][j][1])[k]
+      f = gens(affine_refinements(C)[X][j][1])[k]
     end
-    g = one(base_ring(OO(X)))
+    g = one(ambient_ring(X))
     if m != 0
-      g = gens(affine_refinements(C)[C[l]][m][1])[n]
+      g = gens(affine_refinements(C)[X][m][1])[n]
     end
-    W = SpecOpen(X, [f*g])
+    W = PrincipalOpenSubset(X, [f*g])
```

Comment:
Let's try to find a test for this line, tomorrow. Also, we should run the whole method on covered schemes that use both types of glueings.

Reply:
We can write it when #1573 is merged.